src/Bones/babylon.bone.ts

```diff
@@ -37,12 +41,13 @@ module BABYLON {
             }
         }
 
-        constructor(public name: string, skeleton: Skeleton, parentBone: Bone = null, matrix?: Matrix, restPose?: Matrix) {
+        constructor(public name: string, skeleton: Skeleton, parentBone: Bone = null, localMatrix?: Matrix, restPose?: Matrix, baseMatrix?: Matrix, index?: number) {
             super(name, skeleton.getScene());
             this._skeleton = skeleton;
-            this._localMatrix = matrix ? matrix : Matrix.Identity();
-            this._baseMatrix = this._localMatrix.clone();
+            this._localMatrix = localMatrix ? localMatrix : Matrix.Identity();
             this._restPose = restPose ? restPose : this._localMatrix.clone();
+            this._baseMatrix = baseMatrix ? baseMatrix : this._localMatrix.clone();
+            this._index = index === undefined ? undefined : index;

```

Comment:
index === undefined is not the same as -1. undefined means it will auto calculate the index as it does before this change. -1 means it will not store the transform into the transformMatrices.